templates/infra/kafka.tmpl.yaml

```diff
@@ -72,31 +85,46 @@ spec:
             cpu: {{ .kafka_cpu }}
             memory: {{ .kafka_mem }}
         volumeMounts:
-        - name: data
+        - name: kafka-data
           mountPath: /opt/kafka/data
-      dnsPolicy: ClusterFirst
-      restartPolicy: Always
-      terminationGracePeriodSeconds: 30
       volumes:
-      - name: data
+      - name: kafka-data
         persistentVolumeClaim:
-          claimName: kafka-pv-claim
+          claimName: kafka-pvc
 ---
 apiVersion: v1
 kind: PersistentVolumeClaim
 metadata:
-  name: kafka-pv-claim
-  labels:
-    app: kafka
+  name: kafka-pvc
 spec:
-  accessModes:
-    - ReadWriteOnce
+  accessModes: [ "ReadWriteOnce" ]
   resources:
     requests:
-      storage: 1Gi
+      storage: 10Gi
+  {{- if .persistent_volumes}}
   selector:
     matchLabels:
-      volume: kafka
+      volume: kafka-pv
+  {{- end}}
+  {{- if .storage_class_name}}
+  storageClassName: {{ .storage_class_name }}
+  {{- end}}
+---
+{{- if .persistent_volumes}}
```

Comment:
I don't think making persistent volumes optional really addresses the reason I introduced "volumeClaimTemplates" in my original PR. There are two problems with persistent volumes:

1) they are pain to deal with in a real cloud. You have to manually create them, format the disk, etc
2) they *only* work if replicas==1. Things like kafka, mysql, and zookeeper are horizontally scalable and volumeClaimTemplates allow this.

Reply:
This could certainly be improved. The thinking here was that `storage_class_name` could handle automatic provisioning with a relevant storage class provider, while `persistent_volumes` could handle static provisioning.